src/components/QueryOutput/QueryOutput.test.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import _cloneDeep from 'lodash.clonedeep'
+import { findByTestAttr } from '../../../test/testUtils'
+import { QueryOutput } from '../..'
+import { QueryOutput as QueryOutputWithoutTheme } from '../../components/QueryOutput/QueryOutput'
+import testCases from '../../../test/responseTestCases'
+
+const defaultProps = QueryOutput.defaultProps
+
+const setup = (props = {}, state = null) => {
+  const setupProps = { ...defaultProps, ...props }
+  const wrapper = shallow(<QueryOutput {...setupProps} />).dive()
+  if (state) {
+    wrapper.setState(state)
+  }
+  return wrapper
+}
+
+describe('test each response case', () => {
+  for (let i = 0; i < testCases.length; i++) {
+    describe(`renders correctly: response index ${i}`, () => {
+      test('renders correctly with only token prop', () => {
+        const wrapper = shallow(
+          <QueryOutput queryResponse={testCases[i]} />
+        ).dive()
+        const responseComponent = findByTestAttr(
+          wrapper,
+          'query-response-wrapper'
+        )
+        expect(responseComponent.exists()).toBe(true)
+      })
+      test(`renders correctly with default props: response index ${i}`, () => {
+        const wrapper = setup({
+          queryResponse: testCases[i],
+        })
+        expect(wrapper.exists()).toBe(true)
+      })
+    })
+
+    // describe('props', () => {
+    //   test('does not throw warning with expected props', () => {
+    //     checkProps(QueryOutput, defaultProps)
+    //   })
+    //   describe('showMask', () => {
+    //     test('showMask false does not show mask on drawer open', () => {})
+    //     test('showMask true shows mask on drawer open', () => {})
+    //   })
+    // })
```

Comment:
I was reading up on some guiding principles of FE unit testing and apparently tests like these aren't actually required or really all that useful.

according to [this doc](https://testing-library.com/docs/guiding-principles/), and others like it, we should be testing how the user interacts with the UI, rather than it's underlying functionality.